paddle/fluid/pybind/pybind.cc

```diff
@@ -150,22 +150,47 @@ PYBIND11_PLUGIN(core) {
             LoD new_lod;
             new_lod.reserve(lod.size());
             std::copy(lod.begin(), lod.end(), std::back_inserter(new_lod));
-            new (&instance) LoDTensor(new_lod);
+            new (&instance) LoDTensor(ConvertToOffsetBasedLoD(new_lod));
           })
       .def("__init__", [](LoDTensor &instance) { new (&instance) LoDTensor(); })
       .def("set_lod",
            [](LoDTensor &self, const std::vector<std::vector<size_t>> &lod) {
+             // the input lod info is offset-based
              LoD new_lod;
              new_lod.reserve(lod.size());
              std::copy(lod.begin(), lod.end(), std::back_inserter(new_lod));
              self.set_lod(new_lod);
            })
-      .def("lod", [](LoDTensor &self) -> std::vector<std::vector<size_t>> {
-        auto lod = self.lod();
-        std::vector<std::vector<size_t>> new_lod;
-        new_lod.reserve(lod.size());
-        std::copy(lod.begin(), lod.end(), std::back_inserter(new_lod));
-        return new_lod;
+      .def("set_recursive_sequence_lengths",
+           [](LoDTensor &self, const std::vector<std::vector<size_t>> &lod) {
+             // the input lod info is length-based
+             LoD new_lod;
+             new_lod.reserve(lod.size());
+             std::copy(lod.begin(), lod.end(), std::back_inserter(new_lod));
+             self.set_lod(ConvertToOffsetBasedLoD(new_lod));
+           })
+      .def("lod",
+           [](LoDTensor &self) -> std::vector<std::vector<size_t>> {
```

Comment:
Done. 

BTW, because cpplint would give the following error 
"paddle/fluid/pybind/pybind.cc:620:  Small and focused functions are preferred: PYBIND11_PLUGIN() has 502 non-comment lines (error triggered by exceeding 500 lines).  [readability/fn_size] [1]" 
if I add warning messages. In order to pass the pre-commit check, I deleted the `has_valid_lod` API code in `pybind.cc` because it is not an very important feature and may be added in future PR after the `pybind.cc` has been refactored to be smaller code.

Reply:
Is it possible to use ```// NOLINT``` to disable this lint error?